add Render to Deployment Guides; close #3683 (#3684)
hasura/graphql-engine: docs/graphql/manual/guides/deployment/render.rst

> Hasura GraphQL Engine on Render

Reviewer: ```suggestion
Deploying Hasura GraphQL engine on Render
```

> ====================================================================

Reviewer: The underlines should have the same length as the titles.

.. contents:: Table of contents
  :backlinks: none
  :depth: 1
  :local:

This guide talks about how to deploy the Hasura GraphQL engine on `Render
<https://render.com>`__.

Pre-requisites
--------------

> - GitHub (or alternative) account

Reviewer: What is meant by "alternative"?

Author: Gitlab, mostly, but possibly Bitbucket as well. I didn't take the time to confirm it is the same process. We could leave the bit about alternates out.

Reviewer 2: `GitHub` or `GitLab` account.

- Render account associated with a Github account and with billing enabled

Create a repository
-------------------

> Create a new GitHub repository

Reviewer: Can we have bullet points on the instructions here.

Author: Attempted fix, but I'm worried I didn't do this right.

Within the new repository, create a new file named ``Dockerfile`` that 

> references the latest Hasura GraphQL Engine docker image by including the 

Reviewer: ```suggestion
references the latest Hasura GraphQL engine Docker image by including the 
```

following single line in the file:

.. code-block:: bash

   FROM hasura/graphql-engine:latest

> Using Render Infrastrcuture as Code (Fastest)

Reviewer: Not sure what is meant by "using Render infrastructure as code"

Author: This is a capabiltiy of Render. I'm not sure if we should refer to it as a product name "Infrastructure as Code" (capitalized proper noun) or just a feature description (lowercase)

Reviewer 2: I think we can change this to Using `render.yaml` (fastest)

---------------------------------------------

> In the same repository, create a second new file, this one named 

Reviewer: ```suggestion
In the same repository, create a second new file named 
```

> ``render.yaml``. Includes the following Render Infrastrcuture as 

Reviewer: ```suggestion
``render.yaml``. Insert the following Render infrastructure as 
```

> Code configuration:

Reviewer: ```suggestion
code configuration:
```

.. code-block:: yaml

   services:
   - type: web
     name: hasura
     env: docker
     healthCheckPath: /
     envVars:
     - key: HASURA_GRAPHQL_DATABASE_URL
       fromDatabase:
         name: hasura
         property: connectionString
     - key: HASURA_GRAPHQL_ENABLE_CONSOLE
       value: true
   databases:
   - name: hasura

> Within the Render Dashboard, find the YAML section and the New from 

Reviewer: ```suggestion
On the Render dashboard, find the **YAML** section and the **New from 
```

> YAML button or `go directly there <https://dashboard.render.com/select-repo?type=iac>`_ 

Reviewer: ```suggestion
YAML** button or `go directly there <https://dashboard.render.com/select-repo?type=iac>`_ 
```

and select the repository you've created.

> This it! Skip to notes about the Hasura Console and debugging with logs below.

Reviewer: ```suggestion
That's it! Skip to notes about the Hasura console and debug with the logs below.
```

> This it! Skip to notes about the Hasura Console and debugging with logs below.

Reviewer: What is meant by "skip to notes about the Hasura console"?

Author: Skip to the next section titled "Provision the Database and Web Service Manually (Alternate Method)" and proceed to the following section titled "Open the Hasura Console"

> Provision the Database and Web Service Manually (Alternate Method)

Reviewer: ```suggestion
Provision the database and web service manually (alternative method)
```

------------------------------------------------------------------

.. note::

>    If you already have a Render database setup, you can skip this step.

Reviewer: ```suggestion
   If you already have a Render database setup, you can skip the first step.
```

> Create a Postgres server by providing a name in Render's `New Database 

Reviewer: Can we have bullet points for the instructions here?

configuration page <https://dashboard.render.com/new/database>`__.

> Once created, you'll see an Internal Connection String which 

Reviewer: ```suggestion
Once created, you'll see an internal connection string which 
```

is the database URL used in the next step.

Create a web service by selecting the repository you just created in the `New Web 
Service configuration page <https://dashboard.render.com/select-repo?type=web>`__. 
You have to specify a name.

> Upon creation of the web service, find the Environment tab and create two 

Reviewer: ```suggestion
Upon creation of the web service, find the environment tab and create two 
```

environment variables: one named ``HASURA_GRAPHQL_DATABASE_URL`` whose value is 

> the Internal Connection String from the database you created earlier; 

Reviewer: ```suggestion
the internal connection string from the database you created earlier, 
```

> another named ``HASURA_GRAPHQL_ENABLE_CONSOLE`` whose value is ``true``.

Reviewer: ```suggestion
and another named ``HASURA_GRAPHQL_ENABLE_CONSOLE`` whose value is ``true``.
```

Open the Hasura Console
-----------------------

> That's it! You can monitor the deployment of the web service from the Logs tab. 

Reviewer: ```suggestion
That's it! You can monitor the deployment of the web service from the logs tab. 
```

> There you can see error messages if deployment fails. If deployment succeeds, you 

Reviewer: ```suggestion
There you can see error messages if the deployment fails. If the deployment succeeds, you 
```

follow the link at the top of the web service detail page or visit the following 
URL for the Hasura console:

.. code:: 

   http://<web-service-name>.onrender.com

Replace ``<web-service-name>`` with the name specified earlier.

You can create tables and test your GraphQL queries here. Check out :ref:`Making

> your first GraphQL Query <first_graphql_query>` for a detailed guide.

Reviewer: ```suggestion
your first GraphQL Query <first_graphql_query>` for a detailed example.
```

Tearing down
------------

> To clean-up, delete the web service and the database from the Render Dashboard.

Reviewer: ```suggestion
To clean up, delete the web service and the database from the Render dashboard.
```
